core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -163,11 +163,15 @@ public void validateServiceRoles(StackDto stack, String instanceGroup, int scali
         }
     }
 
-    public void validateClusterStatusForStartHostGroup(StackView stack) {
-        Long clusterId = stack.getClusterId();
-        if (clusterId != null && !(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
+    public void validateClusterStatusForStartHostGroup(StackDto stackdto) {
+        Long clusterId = stackdto.getStack().getClusterId();
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stackdto.getBlueprint().getBlueprintText());
+        List<String> unhealthyHostsNames = processor.getUnhealthyDependentHosts(stackdto);
+        if (clusterId != null && !(stackdto.getStack().isAvailable() || stackdto.getStack().isAvailableWithStoppedInstances())) {
             throw new BadRequestException(format("Data Hub '%s' has '%s' state. Node group start operation is not allowed for this state.",
-                    stack.getName(), stack.getStatus()));
+                    stackdto.getStack().getName(), stackdto.getStack().getStatus()));
+        } else if (!unhealthyHostsNames.isEmpty()) {
+            throw new BadRequestException(format("Upscaling is Not Allowed as Data hub '%s' has Unhealthy Nodes ", stackdto.getStack().getName()));
```

Comment:
Same comment here. Also, possibly create a dedicated method for this which checks if the dependentHosts are unhealthy